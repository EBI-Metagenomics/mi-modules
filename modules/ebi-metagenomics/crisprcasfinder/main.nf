process CRISPRCASFINDER {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'microbiome-informatics/genomes-pipeline.crisprcasfinder:4.3.2':
        'microbiome-informatics/genomes-pipeline.crisprcasfinder:4.3.2' }"

    input:
    tuple val(meta), path(fasta)

    output:
    tuple val(meta), path("crisprcasfinder_results/GFF/*.gff"), emit: gff
    tuple val(meta), path("crisprcasfinder_results/TSV"), emit: tsv
    tuple val(meta), path("crisprcasfinder_results/rawCRISPRs.fna"), optional: true, emit: fna
    path "versions.yml" , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    CRISPRCasFinder.pl -i $fasta \
        ${args} \\
        -so /opt/CRISPRCasFinder/sel392v2.so \
        -outdir crisprcasfinder_results

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        CRISPRCasFinder: \$(echo \$(CRISPRCasFinder.pl -v) | grep -o "version [0-9.]*" | sed "s/version //g")
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    mkdir -p crisprcasfinder_results/TSV
    mkdir -p crisprcasfinder_results/GFF
    touch crisprcasfinder_results/GFF/${prefix}_crisprcasfinder.gff
    touch crisprcasfinder_results/rawCRISPRs.fna

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        CRISPRCasFinder: \$(echo \$(CRISPRCasFinder.pl -v) | grep -o "version [0-9.]*" | sed "s/version //g")
    END_VERSIONS
    """
}
