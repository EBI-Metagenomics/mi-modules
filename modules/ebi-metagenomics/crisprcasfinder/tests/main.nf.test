nextflow_process {

    name "Test Process CRISPRCASFINDER"
    script "../main.nf"
    process "CRISPRCASFINDER"

    tag "modules"
    tag "modules_nfcore"
    tag "crisprcasfinder"

    test("test_crisprcasfinder") {
        when {
            process {
                 """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${moduleDir}/tests/fixtures/test.fasta", checkIfExists: true)
                    ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("test_crisprcasfinder - stub") {
        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${moduleDir}/tests/fixtures/test.fasta", checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
