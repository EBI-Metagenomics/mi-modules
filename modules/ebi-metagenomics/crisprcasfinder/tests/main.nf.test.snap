{
    "test_crisprcasfinder": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "CP014688.gff:md5,b7e7c93da5aab77cd20bc29d90d76efe"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "CRISPR-Cas_summary.tsv:md5,b4627cbdf24569918812aa93558246ff",
                            "CRISPR-Cas_summary.xls:md5,b4627cbdf24569918812aa93558246ff",
                            "Crisprs_REPORT.tsv:md5,c094638a9de112ed3d9fbff19c7cc1b3",
                            "Crisprs_REPORT.xls:md5,c094638a9de112ed3d9fbff19c7cc1b3",
                            "crisprs_orientations_count.tsv:md5,f3f46a84feb86977076538e1584efd37"
                        ]
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "rawCRISPRs.fna:md5,c37e1668c2dc913a93bf3831ab0001d8"
                    ]
                ],
                "3": [
                    "versions.yml:md5,978e39de4c8c4593c80b89636cd6363c"
                ],
                "fna": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "rawCRISPRs.fna:md5,c37e1668c2dc913a93bf3831ab0001d8"
                    ]
                ],
                "gff": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "CP014688.gff:md5,b7e7c93da5aab77cd20bc29d90d76efe"
                    ]
                ],
                "tsv": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "CRISPR-Cas_summary.tsv:md5,b4627cbdf24569918812aa93558246ff",
                            "CRISPR-Cas_summary.xls:md5,b4627cbdf24569918812aa93558246ff",
                            "Crisprs_REPORT.tsv:md5,c094638a9de112ed3d9fbff19c7cc1b3",
                            "Crisprs_REPORT.xls:md5,c094638a9de112ed3d9fbff19c7cc1b3",
                            "crisprs_orientations_count.tsv:md5,f3f46a84feb86977076538e1584efd37"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,978e39de4c8c4593c80b89636cd6363c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-03T14:52:34.044725"
    },
    "test_crisprcasfinder - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_crisprcasfinder.gff:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            
                        ]
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "rawCRISPRs.fna:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    "versions.yml:md5,978e39de4c8c4593c80b89636cd6363c"
                ],
                "fna": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "rawCRISPRs.fna:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "gff": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_crisprcasfinder.gff:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "tsv": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,978e39de4c8c4593c80b89636cd6363c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-03T14:52:41.127804"
    }
}